server_settings/test/server_settings/set/set_server_settings_method_test.dart

```diff
@@ -0,0 +1,128 @@
+    settings: TMailServerSettingOptions(alwaysReadReceipts: true),
+  );
+
+  group('set server settings method', () {
+    test('should return expected result', () async {
+      // arrange
+      final baseOption = BaseOptions(method: 'POST');
+      final dio = Dio(baseOption)..options.baseUrl = 'http://domain.com/jmap';
+      final dioAdapter = DioAdapter(dio: dio);
+      dioAdapter.onPost(
+        '',
+        (server) => server.reply(
+          200,
+          {
+            "sessionState": "abcdefghij",
+            "methodResponses": [
+              [
+                "Settings/set",
+                {
+                  "accountId": "123",
+                  "newState": "1",
+                  "updated": {"singleton": {}}
+                },
+                "c0"
+              ],
+              [
+                "Settings/get",
+                {
+                  "accountId": "123",
+                  "notFound": [],
+                  "state": "1",
+                  "list": [
+                    {
+                      "id": "singleton",
+                      "settings": {"read.receipts.always": "true"}
+                    }
+                  ]
+                },
+                "c1"
+              ]
+            ]
+          },
+        ),
+        data: {
+          "using": [
+            "urn:ietf:params:jmap:core",
+            "com:linagora:params:jmap:settings"
+          ],
+          "methodCalls": [
+            [
+              "Settings/set",
+              {
+                "accountId": "123",
+                "update": {
+                  "singleton": {
+                    "id": "singleton",
+                    "settings": {"read.receipts.always": "true"}
+                  }
+                }
+              },
+              "c0"
+            ],
+            [
+              "Settings/get",
+              {
+                "accountId": "123",
+                "ids": ["singleton"]
+              },
+              "c1"
+            ]
+          ]
+        },
+        headers: {
+          "accept": "application/json;jmapVersion=rfc-8621",
+          "content-length": 503
+        },
+      );
+
+      final httpClient = HttpClient(dio);
+      final processingInvocation = ProcessingInvocation();
+      final accountId = AccountId(Id('123'));
+
+      final setServerSettingsMethod = SetServerSettingsMethod(accountId)
+        ..addUpdatesSingleton({
+          ServerSettingsIdSingleton.serverSettingsIdSingleton.id:
+              expectedResult,
+        });
+      final requestBuilder =
+          JmapRequestBuilder(httpClient, processingInvocation)
+            ..invocation(setServerSettingsMethod);
+      final getServerSettingsMethod = GetServerSettingsMethod(accountId)
+        ..addIds({ServerSettingsIdSingleton.serverSettingsIdSingleton.id});
+      final getServerSettingsInvocation =
+          requestBuilder.invocation(getServerSettingsMethod);
```

Comment:
don't use IDE format code. Indent should be 2 space

Reply:
done